File: Javascript/__tests__/stack-queue/Stack.js
class Stack {
  constructor() {
    this._storage = [];
  }

  push(val) {
    return this._storage.push(val);
  }

  pop() {
    return this._storage.pop();
  }

  peek() {
    const length = this._storage.length;
    return length > 0 ? this._storage[length - 1] : null;
  }

  top() {
    return this.peep();
  }

  empty() {
    return this._storage.length === 0;
  }

  size() {
    return this._storage.length;
  }
}

File: swift/leetcode.playground/Sources/Utils/Stack.swift
import Foundation

public struct Stack<T> {
  private var storage: [T] = []

  public mutating func push(_ val: T) {
    storage.append(val)
  }

  public mutating func pop() -> T? {
    guard let last = storage.last else {
      return nil
    }
    storage = storage.dropLast()
    return last
  }

  public var top: T? {
    return storage.last
  }

  public var isEmpty: Bool {
    return storage.count == 0
  }
}
